# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1321522229 0
#      Thu Nov 17 09:30:29 2011 +0000
# Branch gui
# Node ID 5336db83b935f7310e66facb87d579907286aa15
# Parent  bdad80f28d5c88fb18e0b5d833dbb5e303c2ca5a
[Win32] Force hiding console window.

* src/terminal/win32/QConsole.cpp (QConsolePrivate::QConsolePrivate):
Force hiding the console window, in case it wasn't created hidden.

diff --git a/gui/src/terminal/win32/QConsole.cpp b/gui/src/terminal/win32/QConsole.cpp
--- a/gui/src/terminal/win32/QConsole.cpp
+++ b/gui/src/terminal/win32/QConsole.cpp
@@ -258,16 +258,19 @@ QConsolePrivate::QConsolePrivate (QConso
 
   hStdOut = GetStdHandle (STD_OUTPUT_HANDLE);
 
   log ("Console allocated: hStdOut: %p\n", hStdOut);
 
   m_stdOut = hStdOut;
   m_consoleWindow = GetConsoleWindow ();
 
+  // In case the console window hasn't been created hidden...
+  ShowWindow (m_consoleWindow, SW_HIDE);
+
   CONSOLE_SCREEN_BUFFER_INFO sbi;
 
   GetConsoleScreenBufferInfo (hStdOut, &sbi);
   m_bufferSize = QSize (sbi.dwSize.X, qMax (sbi.dwSize.Y, (SHORT)500));
   m_consoleRect = QRect (sbi.srWindow.Left, sbi.srWindow.Top,
                          sbi.srWindow.Right - sbi.srWindow.Left + 1,
                          sbi.srWindow.Bottom - sbi.srWindow.Top + 1);
   m_cursorPos = QPoint (sbi.dwCursorPosition.X, sbi.dwCursorPosition.Y);
